various fixes and features: synchronize rules from local dir, check service status after restart, rename port ansible var, add become where needed allow use of SSL elasticsearch cluster connection

NanoPish opened the pull request:
* automatic rules upload -> local dir {{ elastalert_local_rules_dir }}
* automatic rules upload: delete rules on server that are not in the {{ elastalert_local_rules_dir }} ansible machine directory (=> synchronize ~= rsync)
* automatic service status check after restart -> see handlers
* add become: yes everywhere it is needed to avoid unecessary become
* rename {{ port }} to {{ es_port }} to make it clearer
* SSL elasticsearch connection

Commit:
* automatic rules upload -> local dir {{ elastalert_local_rules_dir }}
* automatic rules upload: delete rules on server that are not in the {{ elastalert_local_rules_dir }} ansible machine directory (=> synchronize ~= rsync)
* automatic service status check after restart -> see handlers
* add become: yes everywhere it is needed to avoid unecessary become
* rename {{ port }} to {{ es_port }} to make it clearer
* SSL elasticsearch connection

NanoPish commented:
Hey, are you interested in those features?

iamabhishek-dubey commented on vars/main.yml:
```diff
@@ -1,9 +1,10 @@
 ---
 host_name: localhost
-port: 9300
+es_port: 9300
```
Let's keep it 9200 because 9300 generally is for TCP connections

iamabhishek-dubey reviewed (changes requested):
Just one change

Commit:
document variables
put variables frequently changed by user in defaults and others in vars, as recommended by https://codereviewvideos.com/course/ansible-tutorial/video/variable-precedence-where-to-put-your-role-vars

Commit:
var and documentation

Commit:
rules dir correct var

Commit:
rm useless file
make deleting unmanaged rules optional

Commit:
optional delete rule
correct doc variables

Commit:
readme more example

Commit:
correct variable use

Commit:
correct upload bool

Commit:
remove double when

NanoPish commented:
Hey! I updated the documentation with all the variables related to automatically uploading and deleting rules, using SSL...
I also added examples and version change in the README.md.
I let you tag the release

Commit:
rd companyx

NanoPish commented:
I also tested the playbook with and without rule upload to check everything is okay
tell me if its all good

iamabhishek-dubey commented:
All things look good, can you resolve merge conflicts?

NanoPish committed:
Merge branch 'master' into master

NanoPish commented:
done

NanoPish commented:
good?

iamabhishek-dubey commented:
Thanks for your contribution!!
